config/config.go

```diff
@@ -40,16 +40,18 @@ type Field struct {
 	version     uint64
 	structField CfgType
 	sources     map[Source]string
+	chNotify    chan<- string
```

Comment:
Having the ability to track changes in configuration is awesome, but I think sending just a string doesn't give too much value.
It would be better to send a notification `struct` with more information like the previous and next values along with the field (at least its name) so that some kind of calculation can be done from the user.
For example if there is a bigger change than `x%` or maybe send these values to another service (ex. auditing, event message etc.)

Reply:
Makes sense. On it.